MetaMask/metamask-extension
# Fix for showing warning when no pending confirmations when network switching

## Explanation

Added fix, so now warning flag is not showing when switching network and when there are no pending confirmations in activity tab, it is showing only when there are unapproved confirmations.
* Fixes #16646

## Screenshots/Screencaps
### Before

![pendingConfirmationsBug](https://user-images.githubusercontent.com/120601632/212329371-1bfb2145-56dc-4fd0-a914-aba9c2e04f97.gif)


### After

![pendingConfirmationsBugFix](https://user-images.githubusercontent.com/120601632/212329398-1862df7e-43fa-46cd-ab31-36966d09f498.gif)

## Manual Testing Steps

1. Check in Metamask extension that there are no unapproved transactions in Activity tab
2. Go to https://etherscan.io/token/0xdac17f958d2ee523a2206206994597c13d831ec7#writeContract
3. Click on Connect to  Web3
4. Click Ok
5. Click on Metamask tab
6. Now the warning flag wont show up
